backend/src/v2/component/launcher_v2.go

```diff
@@ -178,24 +178,26 @@ func (l *LauncherV2) Execute(ctx context.Context) (err error) {
 	var outputArtifacts []*metadata.OutputArtifact
 	status := pb.Execution_FAILED
 	defer func() {
-		if perr := l.publish(ctx, execution, executorOutput, outputArtifacts, status); perr != nil {
+		if err == nil {
```

Comment:
Does this still publish failures? If `err` is `nil`, then I think status is always `pb.Execution_COMPLETE` due to it meaning that `executeV2` did not return an error.

Reply:
Yeah, I haven't noticed that.
I'll revert it.

Reply:
Done.